### DISCUSSION TITLE
gatsby-source-wordpress get all woocommerce products

### DISCUSSION THREAD

**Discussion by danny-wood** [gatsbyjs/gatsby#30867] [Help]
I have a WP site with the following plugins installed: 
WP GraphQL - 1.3.5
WPGraphQL WooCommerce (WooGraphQL) - 0.8.1

How do I get all products (around 500) into my Gatsby site when using gatsby-source-wordpress (5.2.3). I can only get about 20 products, if I change the perPage property from 20 to 100 I get an error:

> Your WordPress server is either overloaded or encountered a PHP error.
> Error occurred while fetching nodes of the "Product" type.

Here's my config:

```js
{
      resolve: `gatsby-source-wordpress`,
      options: {
        url: process.env.GATSBY_WP_DOMAIN,
        verbose: true,
        schema: {
          timeout: 60000,
          perPage: 20,
          requestConcurrency: 5,
          previewRequestConcurrency: 2,
        },
        develop: {
          hardCacheMediaFiles: true,
        },
        debug: {
          graphql: {
            writeQueriesToDisk: true,
          },
        },
        type: {
          Product: {
            limit: 1000,
          },
        },
      },
    }
```

**Comment by ardiewen** [comment]
The plugin options you mentioned are to control performance when you build your schema, as to not overload your Wordpress backend. By default, the source plugin should be fetching ALL nodes, unless you have a limit set on the type (yours is set above the max number of products you have).

You may want to look at this issue  [wp-graphql-woocommerce#474](https://github.com/wp-graphql/wp-graphql-woocommerce/issues/474) as it seems directly related to what you're experiencing.

The perPage matching the actual number of products you see is just a coincidence. You can validate this with other post types (e.g. try pages, posts, other CPTs).

Try downgrading to wp-graphql-1.2.5 per the last comment and see if it works?
